Lib/test/test_cext/__init__.py

```diff
@@ -0,0 +1,92 @@
+# gh-116869: Build a basic C test extension to check that the Python C API
+# does not emit C compiler warnings.
+
+import os.path
+import shutil
+import subprocess
+import sysconfig
+import unittest
+from test import support
+
+
+SOURCE = os.path.join(os.path.dirname(__file__), 'extension.c')
+SETUP = os.path.join(os.path.dirname(__file__), 'setup.py')
+
+
+# gh-110119: pip does not currently support 't' in the ABI flag use by
+# --disable-gil builds. Once it does, we can remove this skip.
+@unittest.skipIf(support.Py_GIL_DISABLED,
+                 'test does not work with --disable-gil')
```

Comment:
It should work now that https://github.com/python/cpython/pull/116878 landed!

(The comment was a bit inaccurate: it turns out the issue was in `wheel`, not `pip`)

Reply:
That's great! I will write a follow-up to patch also test_cppext